File: src/main/java/kr/co/bookand/backend/bookstore/repository/BookStoreRepositoryCustom.java
package kr.co.bookand.backend.bookstore.repository;

import kr.co.bookand.backend.bookstore.domain.BookStore;
import kr.co.bookand.backend.common.domain.Status;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface BookStoreRepositoryCustom {
    Page<BookStore> findAllBySearch(String search, String theme, String status, Pageable pageable);
}

File: src/main/kotlin/kr/co/bookand/backend/bookstore/repository/BookstoreRepositoryCustom.kt
package kr.co.bookand.backend.bookstore.repository

import kr.co.bookand.backend.account.model.Account
import kr.co.bookand.backend.bookstore.model.Bookstore
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface BookstoreRepositoryCustom {
    fun findAllBySearch(
        account: Account,
        search: String?,
        theme: String?,
        status: String?,
        pageable: Pageable
    ): Page<Bookstore>
}